wear/src/main/java/com/android/gpstest/wear/MainActivity.kt

```diff
@@ -0,0 +1,431 @@
+import androidx.activity.viewModels
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.core.app.ActivityCompat
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.Observer
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.wear.compose.material.*
+import com.android.gpstest.Application.Companion.prefs
+import com.android.gpstest.library.LocationLabelAndData
+import com.android.gpstest.library.data.FixState
+import com.android.gpstest.library.data.LocationRepository
+import com.android.gpstest.library.model.GnssType
+import com.android.gpstest.library.model.SatelliteStatus
+import com.android.gpstest.library.model.SbasType
+import com.android.gpstest.library.ui.SignalInfoViewModel
+import com.android.gpstest.library.util.*
+import com.android.gpstest.wear.theme.GpstestTheme
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import java.util.*
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class MainActivity : ComponentActivity() {
+    private var progressBar: ProgressBar? = null
+    private var userDeniedPermission = false
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val signalInfoViewModel: SignalInfoViewModel by viewModels()
+
+    // Repository of location data that the service will observe, injected via Hilt
+    @Inject
+    lateinit var repository: LocationRepository
+
+    // Get a reference to the Job from the Flow so we can stop it from UI events
+    private var locationFlow: Job? = null
+
+    // Preference listener that will cancel the above flows when the user turns off tracking via service notification
+    private val stopTrackingListener: SharedPreferences.OnSharedPreferenceChangeListener =
+        PreferenceUtil.newStopTrackingListener ({ gpsStop() }, prefs)
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        progressBar = findViewById(R.id.progress_horizontal)
```

Comment:
This `progressBar` doesn't do anything because there is no XML layout and you should get an NPE when referencing it.

Instead of doing this, you should create a composable within `WearApp` that is a progress indicator (does Wear have a native one?).

And, you'll need to create another observer in WearApp for getting notified when fix is lost of obtained, like this:

```kotlin
val fixState: FixState by signalInfoViewModel.fixState.observeAsState(FixState.NotAcquired)
```

Reply:
Thanks!